OSDOCS#6723: Included user-defined tags and labels for GCP (#63253)
openshift/openshift-docs: modules/installing-gcp-querying-labels-tags-gcp.adoc

// Module included in the following assemblies:
// * installing/installing_gcp/installing-gcp-customizations.adoc

:_content-type: REFERENCE
[id="installing-gcp-querying-labels-tags-gcp_{context}"]
= Querying user-defined labels and tags for GCP

> After creating the {product-title} cluster, you can access the list of the labels and tags defined for the GCP resources by referring to the new fields, `.status.platformStatus.gcp.resourceLabels` and `.status.platformStatus.gcp.resourceTags` present in the `infrastructures.config.openshift.io/cluster` object as shown in the following sample `infrastructure.yaml` file.

Reviewer: ```suggestion
After creating the {product-title} cluster, you can access the list of the labels and tags defined for the GCP resources  in the `infrastructures.config.openshift.io/cluster` object as shown in the following sample `infrastructure.yaml` file.
```

Author: Ack.

.Sample `infrastructure.yaml` file
[source,yaml]
----
apiVersion: config.openshift.io/v1
kind: Infrastructure
metadata:
 name: cluster
spec:
 platformSpec:
   type: GCP
status:
 infrastructureName: <cluster_id><1>
 platform: GCP
 platformStatus:
   gcp:
     resourceLabels:
     - key: <label_key>
       value: <label_value>
     resourceTags:
     - key: <tag_key_short_name>
       parentID: <OrganizationID/ProjectID>
       value: <tag_value_short_name>
   type: GCP
----

> <1> The cluster ID is the value that is generated at the cluster installation.

Reviewer: ```suggestion
<1> The cluster ID that is generated during cluster installation.
```

Author: Ack.

> <1> The value of `.status.infrastructureName` present in `infrastructures.config.openshift.io/cluster`.

Reviewer: This doesn't make sense. This simply calling out the field name and nothing about what it is.

Author: How about " Contains the value of `.status.infrastructureName` present in `infrastructures.config.openshift.io/cluster`."

Reviewer: The point here is to have callout mention what the field is about. The current statement is redundant.

Author: Confirmed: "The cluster ID is the value that is generated at the cluster installation."

> Along with the user-defined labels, resources have a label defined by the {product-title}. The format of the {product-title} labels is `kubernetes-io-cluster-<cluster_id>:owned`. The clusterID contains the value of `.status.infrastructureName` present in `infrastructures.config.openshift.io/cluster`.

Author: Ack.
